src/Microsoft.Framework.Configuration/ConfigurationSection.cs

```diff
@@ -4,113 +4,124 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using Microsoft.Framework.Configuration.Internal;
 using Microsoft.Framework.Internal;
 
 namespace Microsoft.Framework.Configuration
 {
-    public class ConfigurationSection : IConfiguration
+    public class ConfigurationSection : ConfigurationBase, IConfigurationSection
     {
-        private readonly IList<IConfigurationSource> _sources = new List<IConfigurationSource>();
+        private string _key;
 
-        public ConfigurationSection(IList<IConfigurationSource> sources)
+        public ConfigurationSection([NotNull] IList<IConfigurationSource> sources)
```

Comment:
A `ConfigurationSection` with an empty key doesn't seem natural. Is it only here because it helps keep the `ConfigurationBinder` test cases simple? It might be that those cases are oversimplified now and we need to add one more level of nesting on the keys we use in them.